plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/DeltaLakeSchemaSupport.java

```diff
@@ -471,18 +472,18 @@ public static void verifySupportedColumnMapping(ColumnMappingMode mappingMode)
         }
     }
 
-    public static List<DeltaLakeColumnMetadata> getColumnMetadata(String json, TypeManager typeManager, ColumnMappingMode mappingMode)
+    public static List<DeltaLakeColumnMetadata> getColumnMetadata(String json, TypeManager typeManager, ColumnMappingMode mappingMode, List<String> partitionColumns)
     {
         try {
             ImmutableList.Builder<DeltaLakeColumnMetadata> columns = ImmutableList.builder();
             Iterator<JsonNode> nodes = OBJECT_MAPPER.readTree(json).get("fields").elements();
             while (nodes.hasNext()) {
                 try {
-                    columns.add(mapColumn(typeManager, nodes.next(), mappingMode));
+                    columns.add(mapColumn(typeManager, nodes.next(), mappingMode, partitionColumns));
                 }
                 catch (UnsupportedTypeException e) {
                     // Write operations are denied by unsupported 'variantType' writer feature
-                    log.debug("Skip unsupported column type: %s", e.type());
+                    log.debug("Skip unsupported column type: %s", e.getMessage());
```

Comment:
This change seems unrelated to the current commit.

Reply:
Hmm, yes, I can revert and do exception class change in a separate commit.